src/Microsoft.PowerShell.Commands.Utility/commands/utility/WriteProgressCmdlet.cs

```diff
@@ -107,7 +95,28 @@ protected override
         void
         ProcessRecord()
         {
-            ProgressRecord pr = new(Id, Activity, Status);
+            ProgressRecord pr;
+            if (string.IsNullOrEmpty(Activity))
+            {
+                if (!Completed)
+                {
+                    ThrowTerminatingError(new ErrorRecord(
+                    new ArgumentException("Missing value for mandatory parameter.", nameof(Activity)),
+                    "MissingActivity",
+                    ErrorCategory.InvalidArgument,
+                    Activity));
+                    return;
+                }
+                else
+                {
+                    pr = new(Id);
```

Comment:
it looks like it might be possible to lose the status if the user provided one. the scenario I'm thinking of is something like this:
`1..10|%{ write-progress -id 1 -status "$($_ * 10)% complete"; start-sleep 1 }`